opm/simulators/linalg/WellOperators.hpp

```diff
@@ -0,0 +1,239 @@
+/*
+  Copyright 2016 IRIS AS
+  Copyright 2019, 2020 Equinor ASA
+  Copyright 2020 SINTEF
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WELLOPERATORS_HEADER_INCLUDED
+#define OPM_WELLOPERATORS_HEADER_INCLUDED
+
+#include <dune/istl/operators.hh>
+
+
+namespace Opm
+{
+
+
+//=====================================================================
+// Implementation for ISTL-matrix based operators
+// Note: the classes WellModelMatrixAdapter and
+// WellModelGhostLastMatrixAdapter were moved from ISTLSolverEbos.hpp
+// and subsequently modified.
```

Comment:
i do not think this is of interest in, say, half a year.

Reply:
Actually it is mostly to document where the code came from since the code history will be discontinuous. If you know a way to make the code lines in the new file carry a log history pointing towards the evolution of the code that would be better!

Reply:
git log --follow.

Reply:
that is weird. it's supposed to work across renames and code migration between files.